app/src/main/java/com/github/polypoly/app/ui/menu/SignInActivity.kt

```diff
@@ -40,31 +41,47 @@ import java.util.concurrent.CompletableFuture
 class SignInActivity : ComponentActivity() {
 
     private var firebaseAuth: FirebaseAuth? = null
-    private var mAuthListener: FirebaseAuth.AuthStateListener? = FirebaseAuth.AuthStateListener {}
+    private val mAuthListener: FirebaseAuth.AuthStateListener = FirebaseAuth.AuthStateListener {
+        launchWelcomeIfReady()
+    }
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
 
-        // Global initialization of the database
         initRemoteDB()
         firebaseAuth = FirebaseAuth.getInstance()
         isSignedIn = false
-        if (firebaseAuth!!.currentUser != null) {
-            launchWelcome()
-        }
+
+        launchWelcomeIfReady()
         setContent { SignInContent() }
     }
 
     override fun onResume() {
         super.onResume()
-        firebaseAuth!!.addAuthStateListener(mAuthListener!!)
+        firebaseAuth!!.addAuthStateListener(mAuthListener)
     }
 
     override fun onStop() {
         super.onStop()
-        firebaseAuth!!.removeAuthStateListener(mAuthListener!!)
+        firebaseAuth!!.removeAuthStateListener(mAuthListener)
+    }
+
+    /**
+     * If the user is correctly signed in Firebase, gets the user and launches [WelcomeActivity]
+     */
+    private fun launchWelcomeIfReady() {
+        val user = firebaseAuth?.currentUser
+        if (user != null) {
+            initCurrentUser(user.uid, user.displayName ?: "default")
+
+            val welcomeActivityIntent = Intent(this, WelcomeActivity::class.java)
+            isSignedIn = true
+            startActivity(welcomeActivityIntent)
+            finish()
+        }
     }
 
+    @Deprecated("Prefer to use launchWelcomeIfReady")
```

Comment:
Why deprecate a private function? You could remove it, there's no issue of compatibility here